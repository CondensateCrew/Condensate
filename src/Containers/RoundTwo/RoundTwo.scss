@import '../../css/variables.scss';
@import '../../css/index.scss';

.round-two-main {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  header {
    width: 100%;
  }
  footer {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    width: 100%;
    p {
      width: fit-content;
    }
  }
}

.question-numbers-div {
  display: flex;
  justify-content: center;
    h2 {
      @extend %buttons;
      border-radius: $one-line-border-radius;
      padding: $one-line-padding-space;
      margin: $extra-small-space;
    }
    margin: $medium-space;
}

h2.current-round {
  background: $genius-gradient-mirror;
}

h2.completed-round {
  box-shadow: $inner-shadow;
}

.template-question-section-wrapper {
  background: $primary-gradient;
  .current-question {
    background-color: $genius-color;
    background-image: $genius-gradient;
  }
}

.responses-section {
  margin-top: $small-space;
  background: $primary-gradient;
  box-shadow: $external-shadow;
  border-radius: $big-blocks-border-radius;
  padding: $medium-space;
  header {
    @extend %one-line-block;
    padding: $one-line-padding-space;
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    align-items: center;
    justify-content: center;
    input {
      @extend %regular-input;
      width: 80%;
      margin: 0;
    }
    img {
      width: $medium-text;
      margin-left: $small-space;
      @extend %hover-transform;
    }
  }
}

.responses-div {
  height: 34vh;
  overflow: scroll;
}
