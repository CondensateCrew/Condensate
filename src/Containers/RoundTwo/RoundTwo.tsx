import React, { useState, useEffect, MouseEvent, ChangeEvent, KeyboardEvent } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { AppStore } from 'interfaces';
import { useHistory } from 'react-router-dom';
import { addQuestionTemplates, reverseTime } from 'redux/actions';
import mockQuestionTemplate from 'data/mockQuestionTemplate';
import Instruction from 'Components/Instruction/Instruction';
import TemplateQuestion from 'Components/TemplateQuestion/TemplateQuestion';
import GenerateInsights from 'Components/GenerateInsights/GenerateInsights';
import Header from 'Components/Header/Header';
import Timer from 'Components/Timer/Timer';
import Check from 'assets/check.svg';
import './RoundTwo.scss';

const RoundTwo:React.FC = () => {
  const [ inputValue, setInputValue ] = useState<string>('')
  const [ responses, setResponses ] = useState<string[]>([])
  const [ currentStep, setCurrentStep] = useState<number>(0)
  const [ time, setTime ] = useState<number>(90)

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(addQuestionTemplates(mockQuestionTemplate))
    return () => {
    };
  }, []);
  let history = useHistory();


  useEffect(() => {
    if (timeEnded) {
      if (currentStep >= 2) {
        history.push('/gameboard/round-three')
      }
      setCurrentStep(currentStep + 1)
      dispatch(reverseTime())
    }
  })

  const { timeEnded, questionTemplates } = useSelector((state: AppStore) => ({
    timeEnded: state.timeEnded,
    questionTemplates: state.questionTemplates
  }))

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value)
  }

  const handleInputSubmit = (e: KeyboardEvent<HTMLInputElement>) => {
    if (e.keyCode === 13) {
      setResponses([...responses, inputValue])
      setInputValue('')
    }
  }

  const handleSubmit = (e: MouseEvent<HTMLElement> | KeyboardEvent<HTMLInputElement>) => {
    setResponses([...responses, inputValue])
  }

  let insights = responses.map((insight:string) => {
    return <GenerateInsights text={insight} responses={responses} setResponses={setResponses}/>
  })

  let displayRounds;
    if (currentStep === 0) {
      displayRounds = [
      <h2 id='one' className='question-number-h2 current-round'>1</h2>,
      <h2 id='two' className='question-number-h2'>2</h2>,
      <h2 id='three' className='question-number-h2'>3</h2>
      ]
    } 
    if (currentStep === 1) {
      displayRounds = [
      <h2 id='one' className='question-number-h2 completed-round'>1</h2>,
      <h2 id='two' className='question-number-h2 current-round'>2</h2>,
      <h2 id='three' className='question-number-h2'>3</h2>
      ]
    }
    if (currentStep === 2) {
      displayRounds = [
      <h2 id='one' className='question-number-h2 completed-round'>1</h2>,
      <h2 id='two' className='question-number-h2 completed-round'>2</h2>,
      <h2 id='three' className='question-number-h2 current-round'>3</h2>
      ]
    }

  return (
    <main className='round-two-main'>
    <Header />
    <section className='round-two-section'>
      <div className='instructions-div-wrapper'>
        <div className='question-numbers-div'>
          {displayRounds}
        </div>
      <Instruction/>
      </div>
      <section className='template-question-section-wrapper'>
        <TemplateQuestion templateQuestion={questionTemplates[currentStep]}/>
        <div className='responses-div'>
          {insights}
          <div className='responses-input-div'>
            <input id='responses-input' onKeyDown={handleInputSubmit} onChange={handleChange}
            type='text' value={inputValue} placeholder='Write your question here...'></input>
            <img src={Check} onClick={handleSubmit} alt='submit-checkmark-icon' />
          </div>
        </div>
      </section>
    </section>
    <footer>
      {!timeEnded && <Timer time={time}/>}
    </footer>
    </main>
  )
}
