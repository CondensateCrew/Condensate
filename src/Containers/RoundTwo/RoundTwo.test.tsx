/**
 * @jest-environment jsdom
 */

import React from 'react';
import RoundTwo from './RoundTwo';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: () => ({
    timeEnded: false,
    exampleSentences: [{
      word: 'toy',
      sentence: 'Play with a toy'
    }],
    chosenWords: ['toy']
  }),
  useDispatch: () => mockDispatch
}));

describe('RoundTwo Component', () => {
  beforeEach(() => {
    jest.mock('react-router-dom', () => ({
      useHistory: () => ({
        push: jest.fn(),
      }),
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot when rendering for the first time', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <RoundTwo />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
  describe('Change Events', () => {
    it('should update state when change event occurs', () => {
      const mockEvent = {target: {value: 'What does the fox say?'}};
      const wrapper: any = mount(
        <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
          <RoundTwo />
        </MemoryRouter>
      );

      wrapper.find('input').simulate('change', mockEvent);
      expect(wrapper.find('input').getDOMNode().value).toEqual('What does the fox say?')
    });
  });
});
