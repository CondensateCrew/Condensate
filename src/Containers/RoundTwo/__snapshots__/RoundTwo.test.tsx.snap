// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundTwo Component should match the snapshot when rendering for the first time 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RoundTwo>
      <main
        className="round-two-main"
      >
        <Header>
          <header
            className="round-header"
          >
            <button
              onClick={[Function]}
            >
              <img
                alt="back"
                src="down.svg"
              />
              to dashboard
            </button>
            <h1>
              Condensate
            </h1>
          </header>
        </Header>
        <section
          className="template-question-section-wrapper"
        >
          <ExampleSentence
            exampleSentence="Play with a toy"
          >
            <h3
              className="template-question-h3"
            >
              Play with a toy
            </h3>
          </ExampleSentence>
          <section
            className="responses-section"
          >
            <header>
              <input
                autoComplete="off"
                id="responses-input"
                maxLength={100}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Write your question here..."
                type="text"
                value=""
              />
              <img
                alt="submit-checkmark-icon"
                onClick={[Function]}
                src="check.svg"
              />
            </header>
            <div
              className="responses-div"
            />
          </section>
        </section>
        <footer>
          <div
            className="question-numbers-div"
          >
            <h2
              className="question-number-h2 current-round"
              id="step-0"
              key="0"
            >
              1
            </h2>
            <h2
              className="question-number-h2 "
              id="step-1"
              key="1"
            >
              2
            </h2>
            <h2
              className="question-number-h2 "
              id="step-2"
              key="2"
            >
              3
            </h2>
          </div>
          <Timer
            time={60}
          >
            <p
              className="timer"
            >
              60
            </p>
          </Timer>
        </footer>
      </main>
    </RoundTwo>
  </Router>
</MemoryRouter>
`;
